toolchain/lex/tokenized_buffer.h

```diff
@@ -306,7 +306,7 @@ class TokenizedBuffer : public Printable<TokenizedBuffer> {
     }
     auto set_ident_id(IdentifierId ident_id) -> void {
       CARBON_DCHECK(kind() == TokenKind::Identifier);
-      CARBON_DCHECK(ident_id.index < (2 << PayloadBits));
+      // Overflow is checked in `Lexer::LexFileEnd`.
```

Comment:
Is it worth noting somewhere that this relies on no identifiers being added to the shared value store outside this specific tokenized buffer?

i.e., this should work because the identifier count is limited by the token count. However, if there's ever a second source of identifiers, this breaks.

Reply:
Added a comment to `Lexer::Finalize`.